.github/workflows/build.yaml

```diff
@@ -2,24 +2,193 @@ name: Formating and build
 
 on:
   push:
-    # Run on the main branch
-    branches:
-      - main
   pull_request:
 
 jobs:
-  clippy_check:
+  check:
     runs-on: ubuntu-latest
+    outputs:
+      do-publish: ${{steps.state.outputs.do-publish}}
+      release: ${{steps.state.outputs.release}}
+      version: ${{steps.version.outputs.version}}
+      prerelease: ${{steps.state.outputs.prerelease}}
     steps:
+      - name: Evaluate state
+        id: state
+        shell: bash
+        run: |
+          test -z "${{github.head_ref}}" && echo '::set-output name=do-publish::true'
+          if [[ "${{ github.event.ref }}" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+           echo ::set-output name=release::true
+           echo ::set-output name=release::true
+          elif [[ "${{ github.event.ref }}" =~ ^refs/tags/v.*$ ]]; then
+           echo ::set-output name=release::true
+           echo ::set-output name=prerelease::true
+          fi
       - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-clippy-${{ hashFiles('**/Cargo.lock') }}
       - run: rustup component add clippy
       - uses: actions-rs/clippy-check@v1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           args: --all-features
-      
+
       - name: Run Rustfmt
         run: cargo fmt -- --check
       - name: Cargo build
         run: cargo build
```

Comment:
The `build` job already builds the code anyway, so we might as well remove this step

Reply:
Good catch :) … I replaced it with `cargo check`, so that we are sure the code is "buildable", before we kick of the other build jobs.